variety/VarietyWindow.py

```diff
@@ -2232,6 +2236,27 @@ def _on_continue(button):
         dialog.run()
         dialog.destroy()
 
+    def show_privacy_dialog(self):
+        dialog = PrivacyNoticeDialog()
+        ok = False
+
+        def _on_accept(*args):
+            dialog.destroy()
+            self.dialogs.remove(dialog)
+            ok = True
+
+        def _on_close(*args):
+            if not ok:
+                # At this point we shouldn't have much to clean up yet!
+                sys.exit(1)
+
+        dialog.ui.accept_button.connect("clicked", _on_accept)
```

Comment:
Just FYI: I think for other dialogs I defined the button handlers directly in Glade (click a button and look for the "Signals" tab), but as long as things work, this approach is fine too.